contrib/cloud.google.com/go/pubsub.v1/pubsub.go

```diff
@@ -0,0 +1,59 @@
+package pubsub
+
+import (
+	"context"
+
+	"cloud.google.com/go/pubsub"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Publish publishes a message on the specified topic, waits for publishing to complete, and returns the result.
+// It is functionally equivalent to (*pubsub.Topic).Publish(ctx, msg).Get(ctx), but it also ensures that the datadog
+// tracing metadata is propagated as attributes attached to the published message.
+func Publish(ctx context.Context, t *pubsub.Topic, msg *pubsub.Message) (serverID string, err error) {
+	span, ctx := tracer.StartSpanFromContext(
+		ctx,
+		"pubsub.publish",
+		tracer.ResourceName(t.String()),
+		tracer.SpanType(ext.SpanTypeMessageProducer),
+		tracer.Tag("message_size", len(msg.Data)),
+		tracer.Tag("ordering_key", msg.OrderingKey),
+	)
+	defer func() {
+		span.Finish(tracer.WithError(err))
+	}()
+	if msg.Attributes == nil {
+		msg.Attributes = make(map[string]string)
+	}
+	tracer.Inject(span.Context(), tracer.TextMapCarrier(msg.Attributes))
```

Comment:
Should we log something when this fails? It does return an error. Perhaps `log.Debug` at the very least.

Reply:
I'm of the opinion that libraries should not log unless explicitly provided with a logger. But sure, I can add one, although an error here really should never happen.

Reply:
They won't, unless it's debug mode. We need this. It helps us debug and resolve support cases. Please add it.